mathesar_ui/src/component-library/select/__meta__/Select.test.ts

```diff
@@ -0,0 +1,185 @@
+      options,
+      triggerAppearance: 'default',
+    },
+  });
+
+  // Options are only rendered when the button is clicked.
+  const selectBtn = getByRole('button');
+  await fireEvent.click(selectBtn);
+
+  // Select the second option.
+  let selectOptions = getAllByRole('option');
+  await fireEvent.click(selectOptions[1]);
+
+  // Expect the button to contain the correct label.
+  expect(selectBtn).toHaveTextContent(options[1].label);
+
+  // Open the options list again, so we can update the
+  // reference to the list of options.
+  await fireEvent.click(selectBtn);
+  selectOptions = getAllByRole('option');
+
+  // Expect the selected option to have the correct classes.
+  expect(selectOptions[1]).toHaveClass('selected');
+  expect(selectOptions[1]).toHaveClass('hovered');
+
+  expect(selectOptions[0]).not.toHaveClass('selected');
+  expect(selectOptions[0]).not.toHaveClass('hovered');
+});
+
+test('select option using keyboard', async () => {
+  const options = [
+    { id: 1, label: 'Option 1' },
+    { id: 2, label: 'Option 2' },
+  ];
+
+  const { getByRole, getAllByRole } = render(Select, {
+    props: {
+      options,
+      triggerAppearance: 'default',
+    },
+  });
+
+  // Options are only rendered when the button is clicked
+  // or Enter is pressed on it.
+  const selectBtn = getByRole('button');
+  await fireEvent.keyDown(selectBtn, { key: 'Enter', code: 'Enter' });
+  await fireEvent.keyUp(selectBtn, { key: 'Enter', code: 'Enter' });
+
+  // Move down one option (By default, the first option is selected).
+  await fireEvent.keyDown(selectBtn, { key: 'ArrowDown', code: 'ArrowDown' });
+  await fireEvent.keyUp(selectBtn, { key: 'ArrowDown', code: 'ArrowDown' });
+
+  // Check if the second option has the correct classes.
+  const selectOptions = getAllByRole('option');
+  expect(selectOptions[1]).toHaveClass('hovered');
+  expect(selectOptions[1]).not.toHaveClass('selected');
+
+  // Select the second option.
+  await fireEvent.keyDown(selectBtn, { key: 'Enter', code: 'Enter' });
+  await fireEvent.keyUp(selectBtn, { key: 'Enter', code: 'Enter' });
+
+  expect(selectBtn).toHaveTextContent(options[1].label);
+});
+
+test('hide select dropdown by pressing ESC', async () => {
+  const options = [
+    { id: 1, label: 'Option 1' },
+    { id: 2, label: 'Option 2' },
+  ];
+
+  const { getByRole, queryByText } = render(Select, {
+    props: {
+      options,
+      triggerAppearance: 'default',
+    },
+  });
+
+  // Options are only rendered when the button is clicked.
+  const selectBtn = getByRole('button');
+  await fireEvent.click(selectBtn);
+
+  // KeyPress was not working, but it does the same thing.
+  await fireEvent.keyDown(selectBtn, { key: 'Escape', code: 'Escape' });
+  await fireEvent.keyUp(selectBtn, { key: 'Escape', code: 'Escape' });
+
+  // Use `queryBy` to avoid throwing an error with `getBy`.
+  const selectOption2 = queryByText(options[1].label);
+  expect(selectOption2).not.toBeInTheDocument();
```

Comment:
We can check by querying by role for the listbox instead of checking by text of an option.

The test is for the listbox(select list dropdown) to be hidden. So it makes more sense to test by it, rather than the options within it.

Reply:
Fixed, changed to `queryByRole('listbox');`.